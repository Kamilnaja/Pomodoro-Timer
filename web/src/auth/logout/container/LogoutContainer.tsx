import React from 'react';
import { connect } from 'react-redux';
import { setUserIsLoggedOut } from '../../store/actions/authActions';
import { AuthState } from '../../store/interfaces/authState';
import { LogoutComponent } from '../component/LogoutComponent';

export interface LogoutContainerProps {
  handleLogout: () => void;
  auth: AuthState;
}

class LogoutContainer extends React.Component<LogoutContainerProps> {
  render() {
    return <LogoutComponent handleLogout={this.props.handleLogout}></LogoutComponent>;
  }
}

const mapStateToProps = (state: { auth: AuthState }) => {
  const authState = state.auth;
  return { auth: authState };
};

const mapDispatchToProps = {
  handleLogout: setUserIsLoggedOut,
};

export default connect(mapStateToProps, mapDispatchToProps)(LogoutContainer);
