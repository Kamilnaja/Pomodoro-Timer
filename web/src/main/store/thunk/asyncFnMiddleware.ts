export async function fetchStats(dispatch: any, getState: any) {
  const response = await client.get("/fakeApi/todos");
  dispatch({ type: "todos/todosLoaded", payload: response.todos });
}
