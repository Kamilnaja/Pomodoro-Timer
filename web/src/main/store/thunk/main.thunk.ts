import TodayStatistics from "../../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
  savePomodoro,
  savePomodoroError,
  savePomodoroSuccess,
} from "../actions/actions";
// todo - different versions for prod etc
const API_URL = "http://localhost:8080/pomodoros";

function makePostRequest() {
  return fetch(API_URL, {
    method: "POST",
  });
}

export function makeGetRequest() {
  return fetch(API_URL);
}

function handleErrors(response: any): Promise<any> {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response.json();
}

export function savePomodoroThunk() {
  return (dispatch: Function) => {
    dispatch(savePomodoro());

    return makePostRequest()
      .then(handleErrors)
      .then(() => dispatch(savePomodoroSuccess()))
      .catch((error) => dispatch(savePomodoroError(error)));
  };
}

export function getPomodorosThunk() {
  return (dispatch: Function) => {
    dispatch(getTodayStatistics());

    return makeGetRequest()
      .then(handleErrors)
      .then((payload: TodayStatistics) => {
        console.log(payload);
        return dispatch(getTodayStatisticsSuccess(payload));
      })
      .catch((error) => dispatch(getTodayStatisticsError(error)));
  };
}
