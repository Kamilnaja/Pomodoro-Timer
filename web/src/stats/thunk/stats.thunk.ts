import { handleErrors } from "../../main/store/thunk/main.thunk";
import { initialConfig } from "../../shared/settings/initialConfig";
import TodayStatistics from "../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
} from "../store/actions/stats.actions";

const API_URL = initialConfig.apiUrl;

const makeGetRequest = () => fetch(API_URL + "/pomodoros_done_today");

export const getPomodorosThunk = () => (dispatch: Function) => {
  dispatch(getTodayStatistics());

  return makeGetRequest()
    .then(handleErrors)
    .then((response) => response.json())
    .then((payload: TodayStatistics) => {
      return dispatch(getTodayStatisticsSuccess(payload));
    })
    .catch((error) => dispatch(getTodayStatisticsError(error)));
};
