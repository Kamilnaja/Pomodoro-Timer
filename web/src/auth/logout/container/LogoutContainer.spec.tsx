import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import { createAuthState } from '../../testing/auth.testing.data';
import { Provider } from 'react-redux';
import LogoutContainer from './LogoutContainer';

const mockStore = configureStore([thunk]);

describe('LogoutContainer', () => {
  let store;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      auth: createAuthState,
    });

    component = renderer.create(
      <Provider store={store}>
        <LogoutContainer closeModal={() => {}} />
      </Provider>,
    );
  });

  it('Should match the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  xit('Should close modal', () => {});
});
