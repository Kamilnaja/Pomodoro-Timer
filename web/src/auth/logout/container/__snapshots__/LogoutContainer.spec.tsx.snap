// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutContainer Should match the snapshot 1`] = `
<div
  className="logout"
>
  <p
    className="logout__question"
  >
    Are you sure to logout?
  </p>
  <div
    className="logout__buttons-wrapper"
  >
    <button
      className="logout__button logout__button--no"
      onClick={[Function]}
    >
      No
    </button>
    <button
      className="logout__button logout__button-yes"
      onClick={[Function]}
    >
      Yes
    </button>
  </div>
</div>
`;
